src/backend/latex/simple.rs

```diff
@@ -147,39 +150,71 @@ pub struct ImageGen;
 impl<'a> SimpleCodeGenUnit<Image<'a>> for ImageGen {
     fn gen(image: WithRange<Image<'a>>, out: &mut impl Write) -> Result<()> {
         let WithRange(Image { label, caption, title, alt_text, path, scale, width, height }, _range) = image;
-        let inline_fig = InlineEnvironment::new_figure(label, caption);
-        inline_fig.write_begin(&mut *out)?;
+        includegraphics(out, label, caption, title, alt_text, path, scale, width, height)?;
+        Ok(())
+    }
+}
 
-        if title.is_some() {
-            writeln!(out, "\\pdftooltip{{")?;
-        }
-        if alt_text.is_some() {
-            write!(out, "\\imagewithtext[")?;
-        } else {
-            write!(out, "\\includegraphics[")?;
-        }
+fn includegraphics(out: &mut impl Write, label: Option<WithRange<Cow<'_, str>>>,
+    caption: Option<WithRange<Cow<'_, str>>>, title: Option<Cow<'_, str>>, alt_text: Option<String>,
+    path: impl AsRef<Path>, scale: Option<WithRange<Cow<'_, str>>>, width: Option<WithRange<Cow<'_, str>>>,
+    height: Option<WithRange<Cow<'_, str>>>,
+) -> Result<()> {
+    let inline_fig = InlineEnvironment::new_figure(label, caption);
+    inline_fig.write_begin(&mut *out)?;
 
-        if let Some(WithRange(scale, _)) = scale {
-            write!(out, "scale={}", scale)?;
-        }
-        if let Some(WithRange(width, _)) = width {
-            write!(out, "width={},", width)?;
-        }
-        if let Some(WithRange(height, _)) = height {
-            write!(out, "height={},", height)?;
-        }
+    if title.is_some() {
+        writeln!(out, "\\pdftooltip{{")?;
+    }
+    if alt_text.is_some() {
+        write!(out, "\\imagewithtext[")?;
+    } else {
+        write!(out, "\\includegraphics[")?;
+    }
 
-        if let Some(alt_text) = alt_text {
-            writeln!(out, "]{{{}}}{{{}}}", path.display(), alt_text)?;
-        } else {
-            writeln!(out, "]{{{}}}", path.display())?;
-        }
+    if let Some(WithRange(scale, _)) = scale {
+        write!(out, "scale={}", scale)?;
+    }
+    if let Some(WithRange(width, _)) = width {
+        write!(out, "width={},", width)?;
+    }
+    if let Some(WithRange(height, _)) = height {
+        write!(out, "height={},", height)?;
+    }
 
-        if let Some(title) = title {
-            writeln!(out, "}}{{{}}}", title)?;
-        }
+    write!(out, "]{{{}}}", path.as_ref().display())?;
+
+    if let Some(alt_text) = alt_text {
+        write!(out, "{{{}}}", alt_text)?;
+    }
 
-        inline_fig.write_end(out)?;
+    writeln!(out)?;
+
+    if let Some(title) = title {
+        writeln!(out, "}}{{{}}}", title)?;
+    }
+
+    inline_fig.write_end(out)?;
+    Ok(())
+}
+
+#[derive(Debug)]
+pub struct SvgGen;
+
+impl<'a> MediumCodeGenUnit<Svg<'a>> for SvgGen {
+    fn gen<'b, 'c>(svg: WithRange<Svg<'a>>, config: &Config, stack: &mut Stack<'b, 'c, impl Backend<'b>, impl Write>) -> Result<()> {
```

Comment:
Could the `config` just be accessible through `stack`? Or even make the whole `temp_file` construction something that is unified there. I'm not sure what the best way is to add the error annotations here but I do not think the handling of configuration parameters should be exposed to the backend. It should also not infect so many other parameter lists for a largely unused functionality.

Reply:
The Config is already exposed to the backend for (stateful) code gen units:

https://github.com/oberien/heradoc/blob/318d5564f901ba5d62c2f8ef6b4c409f0b6003ef/src/generator/mod.rs#L143-L146

https://github.com/oberien/heradoc/blob/318d5564f901ba5d62c2f8ef6b4c409f0b6003ef/src/generator/code_gen_units.rs#L49-L54

This is for example used by `Graphviz` graph generation:

https://github.com/oberien/heradoc/blob/318d5564f901ba5d62c2f8ef6b4c409f0b6003ef/src/backend/latex/complex/graphviz.rs#L30-L31

Arguably, that is the only usage of `Config` in code gen units I was able to find. The backend *does* need the config during preamble generation. Currently it should be possible to remove the config from code gen units and move output-file generation to `Stack` (and duplicate the method in `Generator`). However, I'd preefer opening an issue with a separate PR as it's not directly necessary for this PR.

(I changed svg to use `out_dir` instead of `temp_dir`, but we want to remove `temp_dir` anyways somewhere in the future #10)